pkg/azureutils/azure_disk_utils.go

```diff
@@ -1279,6 +1240,68 @@ func UpdateCRIWithRetry(ctx context.Context, informerFactory azdiskinformers.Sha
 	return updatedObj, err
 }
 
+func getOriginalObjFromClient(ctx context.Context, w workflow.Workflow, informerFactory azdiskinformers.SharedInformerFactory, cachedClient client.Client, azDiskClient azdisk.Interface, obj client.Object) (client.Object, error) {
+	var err error
+	var originalObj client.Object
+	objName := obj.GetName()
+
+	w.Logger().V(5).Infof("getting the original object (%s) from clients", objName)
+
+	switch target := obj.(type) {
+	case *azdiskv1beta2.AzVolume:
+		if informerFactory != nil {
+			originalObj, err = informerFactory.Disk().V1beta2().AzVolumes().Lister().AzVolumes(target.Namespace).Get(objName)
+		} else if cachedClient != nil {
+			originalObj = &azdiskv1beta2.AzVolume{}
+			err = cachedClient.Get(ctx, types.NamespacedName{Namespace: target.Namespace, Name: objName}, originalObj)
+		}
+
+		if azDiskClient != nil && err != nil {
+			originalObj, err = azDiskClient.DiskV1beta2().AzVolumes(target.Namespace).Get(ctx, objName, metav1.GetOptions{})
+		}
```

Comment:
In the case if informerFactory, cachedClient, and azDiskClient are all `nil`, this function will return (nil, nil) and get mishandled in the calling function. Should we also return an error when azDiskClient is `nil`?

Reply:
Yes, good catch. And it's better to check if azDiskClient is nil first. If it's nil, we even don't need to Get the original object since we need azDiskClient to Update().